test/SleepLog-test.js

```diff
@@ -0,0 +1,183 @@
+const chai = require('chai');
+const expect = chai.expect;
+
+const SleepLog = require('../src/SleepLog.js');
+const User = require('../src/User.js');
+
+describe('SleepLog', function() {
+  let sleepLog, user1, user2;
+
+  beforeEach(function() {
+    sleepData = [
+      {
+        "userID": 1,
+        "date": "2019/06/15",
+        "hoursSlept": 6.1,
+        "sleepQuality": 2.2
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/16",
+        "hoursSlept": 4.1,
+        "sleepQuality": 3.8
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/17",
+        "hoursSlept": 8,
+        "sleepQuality": 2.6
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/18",
+        "hoursSlept": 10.4,
+        "sleepQuality": 3.1
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/19",
+        "hoursSlept": 10.7,
+        "sleepQuality": 1.2
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/20",
+        "hoursSlept": 9.3,
+        "sleepQuality": 1.2
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/21",
+        "hoursSlept": 7.8,
+        "sleepQuality": 4.2
+      },
+      {
+        "userID": 2,
+        "date": "2019/06/20",
+        "hoursSlept": 7.0,
+        "sleepQuality": 3.9
+      },
+      {
+        "userID": 2,
+        "date": "2019/06/21",
+        "hoursSlept": 6.5,
+        "sleepQuality": 4.0
+      }
+    ];
+    userData = [
+      {
+        "id": 1,
+        "name": "Luisa Hane",
+        "address": "15195 Nakia Tunnel, Erdmanport VA 19901-1697",
+        "email": "Diana.Hayes1@hotmail.com",
+        "strideLength": 4.3,
+        "dailyStepGoal": 10000,
+        "friends": [
+          16,
+          4,
+          8
+        ]
+      },
+      {
+        "id": 2,
+        "name": "Jarvis Considine",
+        "address": "30086 Kathryn Port, Ciceroland NE 07273",
+        "email": "Dimitri.Bechtelar11@gmail.com",
+        "strideLength": 4.5,
+        "dailyStepGoal": 5000,
+        "friends": [
+          9,
+          18,
+          24,
+          19
+        ]
+      }
+    ];
+    user1 = new User(userData[0]);
+    user1.getSleepLog(sleepData, user1.userID);
+    user2 = new User(userData[1]);
+    user2.getSleepLog(sleepData, user2.userID);
+    sleepLog = new SleepLog(sleepData, user1.id);
+  });
+
+  it('should be a function', function() {
+    expect(SleepLog).to.be.a('function');
+  });
+
+  it('should instantiate the SleepLog class', function() {
+    expect(user1.sleepLog).to.be.an.instanceOf(SleepLog);
+    expect(user2.sleepLog).to.be.an.instanceOf(SleepLog);
+  });
+
+  it('should hold a list of sleep log entries', function() {
+    const user1SleepData = sleepData.filter(entry => entry.userID === user1.id);
```

Comment:
this actually seems too dynamic for testing: it copies the logic from the class file, which means it can't really count as an independent test of that logic. I'm thinking we might just want to hard code/check hard coded values within the `SleepLog.log`